File: 735. Asteroid Collision/Asteroid Collision.java
class Solution {
    public int[] asteroidCollision(int[] asteroids) {
         Stack<Integer>st = new Stack<>() ;
        for(int  i: asteroids)
        {
           if(i>0)
           st.push(i);
           else {
               while(!st.isEmpty() && st.peek()>0 && st.peek()< -i)
               st.pop();
               if(!st.isEmpty() && st.peek()>0 && st.peek() ==  -i)
               st.pop();
               else if(!st.isEmpty() && st.peek()>0 && st.peek() >  -i)
               {
                   
               }
               else
               st.push(i);
           }
        }
        int n = st.size();
        int []ans = new int[n];
        for(int i = n-1 ;i>=0;i--)
        {
            ans[i]= st.pop() ;
        }
        return ans ;
    }
}

Time Complexity -> O(n)
Space Complexity ->  O(n) 

File: 735. Asteroid Collision/Asteroid Collision.php
<?php


    class Solution {

    /**
     * @param Integer[] $asteroids
     * @return Integer[]
     */
    function asteroidCollision($asteroids) {
    $stack = [];
    foreach ($asteroids as $asteroid) {
        while (!empty($stack) && end($stack) > 0 && end($stack) < -$asteroid) {
            array_pop($stack);
        }
        if (empty($stack) || $asteroid > 0 || end($stack) < 0) {
            array_push($stack, $asteroid);
        } elseif (end($stack) == -$asteroid) {
            array_pop($stack);
        }
    }
    return $stack;
}

}

?>